### PR TITLE
Use ZooContext instead of environment variable for orca_eager_mode

### PR DESCRIPTION
**Pull request by cyita**

### CODE REVIEW

**Comment by jason-dai** [comment]
LGTM

File: pyzoo/zoo/common/nncontext.py [hunk 1]
```diff
@@ -133,6 +134,21 @@ def log_output(cls, value):
                                  "or init_spark_on_yarn")
         cls._log_output = value
 
+    @property
+    def orca_eager_mode(cls):
```
Comment (jason-dai) [inline pyzoo/zoo/common/nncontext.py:138]: change to `_orca_eager_mode(cls)`

File: pyzoo/zoo/common/nncontext.py [hunk 2]
```diff
@@ -115,6 +115,7 @@ def init_spark_on_yarn(hadoop_conf,
 class ZooContextMeta(type):
 
     _log_output = False
+    _orca_eager_mode = True
```
Comment (jason-dai) [inline pyzoo/zoo/common/nncontext.py:118]: change to `__orca_eager_mode`

File: pyzoo/zoo/common/nncontext.py [hunk 3]
```diff
@@ -133,6 +134,21 @@ def log_output(cls, value):
...
 
+    @property
+    def orca_eager_mode(cls):
+        """
+        Default to True. Needs to be set before initializing SparkContext.
+        """
+        return cls._orca_eager_mode
+
+    @orca_eager_mode.setter
+    def orca_eager_mode(cls, value):
```
Comment (jason-dai) [inline pyzoo/zoo/common/nncontext.py:145]: change to `_orca_eager_mode(cls, value)`

**Comment by jason-dai** [comment]
LGTM

**Comment by cyita** [comment]
Jenkins passed http://10.239.47.210:18888/job/ZOO-PR-Validation/3612/

**Pull request merged**